### PR TITLE
Add Symfony 7 support

### PR DESCRIPTION
**Pull request by TheCadien**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes part of https://github.com/sulu/sulu/issues/7413
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?
Add Symfony 7 support

#### Why?

is cool

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,14 +4,14 @@
     "type": "sulu-bundle",
     "license": "MIT",
     "require": {
-        "php": "^7.2 || ^8.0",
+        "php": "^8.2 || ^8.3",
```
Comment (Prokyonn) [inline composer.json]: Should also match the sulu 2.5 requirements ^8.0/^8.1

Therefore also the CI should run for these php versions

**Review by Prokyonn — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
         "friendsofsymfony/rest-bundle": "^2.6 || ^3.0",
-        "sulu/sulu": "^2.0.4 || ^2.5.10@dev",
-        "symfony/config": "^4.3 || ^5.0 || ^6.0",
-        "symfony/dependency-injection": "^4.3 || ^5.0 || ^6.0",
-        "symfony/framework-bundle": "^4.3 || ^5.0 || ^6.0",
-        "symfony/http-foundation": "^4.3 || ^5.0 || ^6.0",
-        "symfony/http-kernel": "^4.3 || ^5.0 || ^6.0"
+        "sulu/sulu": "^2.5 || ^2.6@dev",
+        "symfony/config": "^5.4 || ^6.0 || ^7.0",
```
Comment (alexander-schranz) [inline composer.json]: if not required I would keep the `^4.4`

File: composer.json [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
-        "php": "^7.2 || ^8.0",
+        "php": "^8.2 || ^8.3",
         "friendsofsymfony/rest-bundle": "^2.6 || ^3.0",
-        "sulu/sulu": "^2.0.4 || ^2.5.10@dev",
-        "symfony/config": "^4.3 || ^5.0 || ^6.0",
-        "symfony/dependency-injection": "^4.3 || ^5.0 || ^6.0",
-        "symfony/framework-bundle": "^4.3 || ^5.0 || ^6.0",
-        "symfony/http-foundation": "^4.3 || ^5.0 || ^6.0",
-        "symfony/http-kernel": "^4.3 || ^5.0 || ^6.0"
+        "sulu/sulu": "^2.5 || ^2.6@dev",
```
Comment (alexander-schranz) [inline composer.json]: if not required I would keep the lower version

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
Check `composer outdated` output and check that things like jackalope are also upgraded else you will not get Symfony 7 compatibility in the test run see output:


Bildschirmfoto 2024-07-19 um 12 11 51

**Pull request merged**